Ardupilog.m

```diff
@@ -38,90 +39,67 @@
 
             % If user pressed "cancel" then return without trying to process
             if all(obj.fileName == 0) && all(obj.filePathName == 0)
-                disp(['File not found: ', pathAndFileName])
                 return
             end
             
             % THE MAIN CALL: Begin reading specified log file
             readLog(obj);
+            
+            % Clear out the (temporary) properties
+            obj.log_data = char(0);
+            obj.log_data_read_ndx = 0;
         end
         
-        function [] = countMsgs(obj)
-        % Open file, count number of headers = number of messages, close file
+        function [] = readLog(obj)
+        % Open file, read all data, count number of headers = number of messages, process data, close file
+
             % Open a file at [filePathName filesep fileName]
-            [obj.fileID, errmsg] = fopen([obj.filePathName, filesep, obj.fileName],'r');
+            [obj.fileID, errmsg] = fopen([obj.filePathName, filesep, obj.fileName], 'r');
             if ~isempty(errmsg) || obj.fileID==-1
                 error(errmsg);
             end
 
-            readsize = 1024*1024; % Block size to read before performing a count
-                        
-            indices = []; % Array to count header-messages
-            carry = []; % In case a read accidentally splits a header-message
-                        
-            while feof(obj.fileID)==0
-                % Read a block of the file
-                batch = [carry fread(obj.fileID,readsize)'];
-                if isempty(batch)
-                    break
-                end
-                % If the ending of the block is inside the header
-                if batch(end)==obj.header1
-                    % Carry the header to the next batch
-                    carry = batch(end);
-                else
-                    carry = [];
-                end
-                % Append the index-count of the header message
-                indices = [indices strfind(batch,[obj.header1 obj.header2])];
-            end
-            
-            % The number of header occurances is the number of messages
-            obj.numMsgs = length(indices);
-            
+            % Define the read-size (inf=read whole file) and read the logfile
+            readsize = inf; % Block size to read before performing a count
+            obj.log_data = fread(obj.fileID, [1, readsize], '*uchar'); % Read the datafile entirely
```

Comment:
I see that you load all of the file onto memory once, to search for the header message. Do you think this is preferred against my method of fragmented loading?
It's less code that way, but for really long files, do you really want to load 100MB onto your memory for a small increase in performance?
Not a biggy in any way.